Source/WebKit/UIProcess/API/glib/WebKitWebExtension.cpp

```diff
@@ -0,0 +1,1023 @@
+            "has-override-new-tab-page",
+            nullptr, nullptr,
+            FALSE,
+            WEBKIT_PARAM_READABLE);
+
+    /**
+     * WebKitWebExtension:has-commands:
+     * 
+     * Whether the #WebKitWebExtension includes commands that users can invoke.
+     * See webkit_web_extension_get_has_commands() for more details.
+     * 
+     * Since: 2.48
+     */
+    properties[PROP_HAS_COMMANDS] =
+        g_param_spec_boolean(
+            "has-commands",
+            nullptr, nullptr,
+            FALSE,
+            WEBKIT_PARAM_READABLE);
+
+    /**
+     * WebKitWebExtension:has-content-modification-rules:
+     * 
+     * Whether the #WebKitWebExtension includes rules used for content modification or blocking.
+     * See webkit_web_extension_get_content_modification_rules() for more details.
+     * 
+     * Since: 2.48
+     */
+    properties[PROP_HAS_CONTENT_MODIFICATION_RULES] =
+        g_param_spec_boolean(
+            "has-content-modification-rules",
+            nullptr, nullptr,
+            FALSE,
+            WEBKIT_PARAM_READABLE);
+
+    g_object_class_install_properties(objectClass, N_PROPERTIES, properties);
+}
+
+WebKitWebExtension* webkitWebExtensionCreate(const HashMap<String, String>& resources, GError** error)
+{
+    HashMap<String, Ref<API::Data>> newResources;
+    for (auto key : resources.keys()) {
+        auto string = resources.get(key);
+        auto data = API::Data::create(string.utf8().span());
+
+        newResources.set(key, data);
+    }
+
+    Ref extension = WebKit::WebExtension::create(WTFMove(newResources));
+
+    if (!extension->errors().isEmpty()) {
+        auto internalError = extension->errors().last().ptr();
+        g_set_error(error, webkit_web_extension_error_quark(),
+            toWebKitWebExtensionError(internalError->errorCode()), internalError->localizedDescription().utf8().data(), nullptr);
+    }
+
+    WebKitWebExtension* object = WEBKIT_WEB_EXTENSION(g_object_new(WEBKIT_TYPE_WEB_EXTENSION, nullptr));
+    object->priv->extension = WTFMove(extension);
+    return object;
+}
+
+WebKitWebExtension* webkitWebExtensionCreateWithBytes(const HashMap<String, std::span<const uint8_t>>& resources, GError** error)
+{
+    HashMap<String, Ref<API::Data>> newResources;
+    for (auto key : resources.keys()) {
+        auto data = API::Data::create(resources.get(key));
+        newResources.set(key, data);
+    }
+
+    Ref extension = WebKit::WebExtension::create(WTFMove(newResources));
+
+    if (!extension->errors().isEmpty()) {
+        auto internalError = extension->errors().last().ptr();
+        g_set_error(error, webkit_web_extension_error_quark(),
+            toWebKitWebExtensionError(internalError->errorCode()), internalError->localizedDescription().utf8().data(), nullptr);
+    }
+
+    WebKitWebExtension* object = WEBKIT_WEB_EXTENSION(g_object_new(WEBKIT_TYPE_WEB_EXTENSION, nullptr));
+    object->priv->extension = WTFMove(extension);
+    return object;
```

Comment:
Do we really need two create functions? They are mostly the same thing. Maybe we could have a single one that receives GBytes instead of String/span and add an API::Data constructor taking a GRefPtr<GBytes> so that we also avoid copies.